torch/csrc/jit/interpreter.cpp

```diff
@@ -0,0 +1,288 @@
+#include "interpreter.h"
+#include "torch/csrc/jit/ir.h"
+#include "torch/csrc/jit/generated/aten_dispatch.h"
+#ifdef WITH_CUDA
+#include "torch/csrc/jit/fusion_compiler.h"
+#endif
+
+namespace torch { namespace jit {
+
+using InputList = const std::vector<at::Tensor> &;
+using OutputList = std::vector<at::Tensor>&;
+using Callback = std::function<void(InputList, OutputList)>;
+// Returns a function implementing functionality of a given node,
+// or nullptr if it's a no-op for autograd.
+Callback getCallback(Node *node) {
+  IR_IFM(node, PythonOp)
+    throw NotImplementedException();
+  IR_ELSEIFM(CppOp)
+    throw NotImplementedException();
+  IR_ELSEIF(Select)
+    barf("getCallback() on select?");
+  IR_ELSEIF(FusionGroup)
+#ifdef WITH_CUDA
+    auto fusion_fn = sharedFusionCompiler().getOrCompile(*value->g(kSubgraph));
+    return [fusion_fn](InputList inputs, OutputList outputs) {
+      fusion_fn->launch(inputs, outputs);
+    };
+#else
+    throw std::runtime_error("don't know how to execute FusionGroups without CUDA");
+#endif
+  IR_ELSEIF(Constant)
+    auto t = value->t(kvalue);
+    return [t](InputList inputs, OutputList outputs) {
+      outputs.push_back(t);
+    };
+  IR_ELSEIF(Undefined)
+    return [](InputList inputs, OutputList outputs) {
+      outputs.push_back(at::Tensor());
+    };
+  IR_ELSE()
+    return getTensorOp(node).op;
+  IR_END()
+}
+
+
+// We need some lists for inputs and outputs. To keep all the memory
+// contiguous we allocate a single vector and use offsets into the vector
+// which are stored in the RegList struct
+// start is an offset into int_data of Function if this list is integers
+// and bool_data of Function if this list is booleans (only free_flags)
```

Comment:
I'm annoyed that there is no static or dynamic information telling me whether or not a RegList struct indexes into int_data or bool_data. Can we at least get comments saying which array we're indexing into?

Reply:
(The overengineer in me wonders if you couldn't use a custom allocator to get everything contiguous... probably not worth it haha)

Reply:
After reading more code, I also wonder if this isn't misnamed. RegList struct is being used to universally represent "array in contiguous sequence of memory", it really isn't a "list of registers" at all.

Reply:
Yeah, we can probably add some more infra to make this clearer.